File: src/models/DiscordAccount.ts
import { DomainObject } from './DomainObject';
import { DEFAULT_DISCORD_USER_AGENT } from '../constants';

/**
 * Discord account model
 */
export class DiscordAccount extends DomainObject {
  /** Guild ID */
  guildId?: string;

  /** Channel ID */
  channelId?: string;

  /** User Token */
  userToken?: string;

  /** User UserAgent */
  userAgent: string = DEFAULT_DISCORD_USER_AGENT;

  /** Is available */
  enable: boolean = true;

  /** Concurrency */
  coreSize: number = 3;

  /** Queue length */
  queueSize: number = 10;

  /** Task timeout (minutes) */
  timeoutMinutes: number = 5;

  /**
   * Get display name
   */
  getDisplay(): string {
    return this.channelId || '';
  }
}

File: src/main/java/com/github/novicezk/midjourney/domain/DiscordAccount.java
package com.github.novicezk.midjourney.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.github.novicezk.midjourney.Constants;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@ApiModel("Discord Account")
public class DiscordAccount extends DomainObject {

	@ApiModelProperty("Guild ID")
	private String guildId;
	@ApiModelProperty("Channel ID")
	private String channelId;
	@ApiModelProperty("User Token")
	private String userToken;
	@ApiModelProperty("User UserAgent")
	private String userAgent = Constants.DEFAULT_DISCORD_USER_AGENT;

	@ApiModelProperty("Is available")
	private boolean enable = true;

	@ApiModelProperty("Concurrency")
	private int coreSize = 3;
	@ApiModelProperty("Queue length")
	private int queueSize = 10;
	@ApiModelProperty("Task timeout (minutes)")
	private int timeoutMinutes = 5;

	@JsonIgnore
	public String getDisplay() {
		return this.channelId;
	}
}
